### DISCUSSION TITLE
Cancellazione e Creazione

### DISCUSSION THREAD

**Discussion by giocampo** [iisgiua/giuaschool#524] [Idee e proposte]
Salve, servirebbe la possibilità di eliminare definitivamente e/o cambiare la tipologia di utente registrato. Un utente inserito precedentemente come docente e successivamente passa al ruolo di segretario, in questo momento devo disabilitarlo con il ruolo 1 e crearlo per il ruolo 2 ma dando altre generalità per poter accedere. Altra possibilità l'utente va eliminato perche non piu presente in istituto, ed in quel caso li l'eliminazione se4rvirebbe per tenere pulita la base di dati. 
Per quanto riguarda la creazione di un docente dopo averlo creato con tutti i dati il modulo scaricato anziche dare accesso  con l'email inserita mette il pattern username@giua.edu.it

**Comment by trinko** [comment]
Gli utenti non possono essere eliminati perché sono legati alle procedure svolte nel registro: da quelle più delicate (per i docenti: gli scrutini, i voti e le firme, per gli alunni le assenze e i voti, ecc.) a quelle più banali, ma non per questo meno importanti, come i log delle operazioni fatte (ad esempio login e logout).

L'unico momento in cui è corretto eliminarli e con la procedura di passaggio d'anno, dove gli utenti disabilitati vengono eliminati, facendo pulizia sul database.

Anche per il cambio ruolo, permettendolo, ci sarebbe lo stesso problema.

**Comment by trinko** [comment]
> Per quanto riguarda la creazione di un docente dopo averlo creato con tutti i dati il modulo scaricato anziche dare accesso con l'email 
> inserita mette il pattern [username@giua.edu.it](mailto:username@giua.edu.it)

Credo che ci sia un problema nella tua configurazione del registro. Controlla i parametri.
